src/actions/route_helpers.cr

```diff
@@ -67,4 +67,16 @@ module DppmRestApi::Actions::RouteHelpers
       end
     end
   end
+
+  # Get a prefix from a context, which has a source name.
+  def get_prefix_with_source_name(context : HTTP::Server::Context) : DPPM::Prefix
+    prefix = DPPM::Prefix.new(
+      path: Actions.prefix.path.to_s,
+      group: Actions.prefix.group,
+      source_name: URI.unescape(context.params.url["source_name"]),
+      source_path: Actions.prefix.source_path
+    )
+    prefix.ensure_pkg_dir
```

Comment:
Since everything returns in crystal, I try to make methods like these chainable (end with returning `self`) when it makes sense (like here) or otherwise mark them as returning nil so that one doesn't unintentionally try to use the value of it. Just a suggestion.

Reply:
By using `.tap &.`?

Reply:
no, like

```
  # Ensure the presence of the `pkg` directory.
  def ensure_pkg_dir
    Dir.mkdir(@pkg.to_s) if !File.exists? @pkg.to_s
    self
  end
```

Reply:
Why we will return `self`? `Nil` is fine, that's a file system operation. Maybe returning `@pkg` but I don't see the use case.

Reply:
I was suggesting to return self because of the use case. Statements like these, which are a guard against an error case almost always are immediately followed by doing something on or returning the value which was just checked by them. Here we do `prefix.ensure_pkg_dir; prefix`, and that is likely to be the pattern used by any code which calls `#ensure_pkg_dir`. Like with the stdlib's `not_nil!`?

Like I said it's just a suggestion, something I picked up from other Crystal projects that I've found to be helpful.

Reply:
I have no strong opinion on this, as long as the return type is documented.
I've used `tap` here to shorten the expression. I may do the change when there will be an opportunity to benefit of it.

Reply:
CI is failing due to this